core/mutex.c

```diff
@@ -58,6 +64,19 @@ int _mutex_lock(mutex_t *mutex, int blocking)
         else {
             thread_add_to_list(&mutex->queue, me);
         }
+
+#ifdef MODULE_CORE_PRIORITY_INHERITANCE
+        thread_t *owner = (thread_t *)thread_get(mutex->owner);
+        /* if my priority is higher than the priority of the current mutex
+         * owner, I lend my priority to the current owner */
+        if (me->priority < owner->priority) {
+            DEBUG("PID[%" PRIkernel_pid "]: changing priority from %i -> %i\n",
+                  owner->pid, owner->priority, me->priority);
+            mutex->prio_before = owner->priority;
```

Comment:
only if `prio_before == THREAD_PRIORITY_UNDEF`. Because if you have three threads you'll screw the real priority of the first. Example:

t1 (low prio): locks
t2 (mid prio): locks, t1->mid, prio_before = low
t3 (high prio): locks, t1->high,prio_before = mid, but should still be the original one